server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -152,74 +160,136 @@ public void setupEach() throws IOException {
     userController = new UserController(db);
   }
 
-  @Test
-  public void getAllUsers() throws IOException {
+  /**
+   * Construct an instance of `ContextUtil`, which is essentially
+   * a mock context in Javalin. See `mockContext(String, Map)` for
+   * more details.
+   */
+  private Context mockContext(String path) {
+    return mockContext(path, Map.of());
```

Comment:
Consider using `Collections.emptyMap()` here (just for readability 🙂)

Reply:
Totally! That would be much better – thanks!